language/move-vm/runtime/src/move_vm.rs

```diff
@@ -21,6 +21,10 @@ impl MoveVM {
         }
     }
 
+    pub fn loader(&self) -> &Loader {
```

Comment:
this feels really really odd....

Reply:
Yes this is indeed a temporary solution -- I'm planning to improve it in a subsequent diff. (My intuition is that I'll either implement the `Session` API I talked about previously or something similar. In other words, the client (Libra VM) really shouldn't be creating a `TransactionDataCache` on its own without resorting to the MoveVM instance.)

Reply:
my take is that it would be good if we did the runtime work (which is here already), keep caches change at a minimum and move the AP/FatType cache from the loader, elsewhere. But I would keep the loader reference inside the VM and maybe no try to expose much for now

Reply:
I should be able to keep the loader reference inside the VM with a simple change. (Adding a `fn new_txn_data_cache(&self) -> TransactionDataCache` to `MoveVM`)